service/src/test/java/com/kodemon/service/test/facade/FightFacadeTest.java

```diff
@@ -0,0 +1,301 @@
+import com.kodemon.api.facade.FightFacade;
+import com.kodemon.persistence.dao.GymDao;
+import com.kodemon.persistence.dao.PokemonDao;
+import com.kodemon.persistence.dao.TrainerDao;
+import com.kodemon.persistence.dao.TrainerFightDao;
+import com.kodemon.persistence.entity.*;
+import com.kodemon.persistence.enums.PokemonName;
+import com.kodemon.persistence.enums.PokemonType;
+import com.kodemon.service.config.ServiceConfig;
+import com.kodemon.service.facade.FightFacadeImpl;
+import com.kodemon.service.implementations.BeanMappingServiceImpl;
+import com.kodemon.service.interfaces.*;
+import com.kodemon.service.util.Pair;
+import org.joda.time.DateTime;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+import java.util.*;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Oliver Roch
+ */
+
+@ContextConfiguration(classes = ServiceConfig.class)
+public class FightFacadeTest extends AbstractTestNGSpringContextTests {
+
+    private BeanMappingService beanMappingService;
+    private TrainerFightService trainerFightService;
+    private PokemonFightService pokemonFightService;
+    private PokemonService pokemonService;
+    private TrainerService trainerService;
+    private BadgeService badgeService;
+    private TimeService timeService;
+    private FightFacade fightFacade;
+
+    private Trainer challenger;
+    private Trainer defender;
+    private UserDTO challengerDTO;
+    private UserDTO defenderDTO;
+    private Gym targetGym;
+    private GymDTO targetGymDTO;
+    private Pokemon pikachu;
+    private Pokemon onix;
+    private PokemonDTO pikachuDTO;
+    private PokemonDTO onixDTO;
+    private FightDTO fight1;
+    private FightDTO fight2;
+    private TrainerFight fight1e;
+    private TrainerFight fight2e;
+    private List<TrainerFight> trainerFights;
+    private List<FightDTO> fights;
+
+    @BeforeClass
+    public void prepare() {
+        beanMappingService = mock(BeanMappingService.class);
+        trainerService = mock(TrainerService.class);
+        trainerFightService = mock(TrainerFightService.class);
+        pokemonFightService = mock(PokemonFightService.class);
+        pokemonService = mock(PokemonService.class);
+        badgeService = mock(BadgeService.class);
+        timeService = mock(TimeService.class);
+        fightFacade = new FightFacadeImpl(beanMappingService, trainerFightService, pokemonFightService, pokemonService,
```

Comment:
I mean that this could be written like this:
```
new FightFacadeImpl(
    beanMappingService,
    trainerService,
    ...);
```
which is more clear I think, but it's up to you